experimental/dds/tree/src/SharedTree.ts

```diff
@@ -1188,6 +1188,30 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 		return internalEdit;
 	}
 
+	/**
+	 * Merges `edits` from `other` into this SharedTree.
+	 * @param other - Tree containing the edits that should be applied to this one.
+	 * @param edits - Array of edits from `other` to apply.
+	 * @param stableIdRemapper - Optional remapper to translate stable identities from `other` into stable identities on this tree.
+	 * Any references that `other` contains to a stable id `foo` will be replaced with references to the id `stableIdRemapper(foo)`.
+	 *
+	 * Payloads on the edits are left intact.
+	 * @returns a list containing `EditId`s for all applied edits.
+	 */
+	public mergeEditsFrom(
+		other: SharedTree,
+		edits: readonly Edit<ChangeInternal>[],
```

Comment:
Do we have to expose `ChangeInternal` here? Is this not just the edit log, which is accessible in the scope of this method?

Reply:
We'd like to remove that type from the public API.

Reply:
Expected usage from Whiteboard would be to exclude the first edit (since it's the setup edit to set initial state). But I could imagine this being used for a contiguous subset of edits.

We could type it to be something other than `ChangeInternal`, but anything else is going to require a cast in consumer code or code here, which is worse IMO. If we fixed `edits` to claim it returned an opaque handle type, we could clean up this API to take that in and then cast with more confidence.

FWIW current changes are what's in ts-pipe on the Whiteboard side, so we do need to eventually expose either a method with this semantics or some of the edit conversion methods. Noah and I generally think the semantics of this are clearer (though it's a little strange to have as another primitive in addition to `applyEdit`)

Reply:
I don't think the opaque handle type thing is actually too bad of a change to make, let me see if I can just clean that up.

Reply:
See #9950 which helps, but doesn't make everything fully typesafe.

Reply:
latest references the opaque handle type added by #9950. Resolving :)